bliss/surveys/dc2.py

```diff
@@ -237,38 +244,47 @@ def from_file(cls, cat_path, wcs, height, width, band):
         star_log_fluxes = star_log_fluxes[x_mask]
         galaxy_params = galaxy_params[x_mask]
         galaxy_fluxes = galaxy_fluxes[x_mask]
+        objid = objid[x_mask]
 
         plocs = plocs[x_mask]
-        plocs_rv = plocs.clone()
-        plocs_rv[:, 0] = plocs[:, 1]
-        plocs_rv[:, 1] = plocs[:, 0]
         source_type = source_type[x_mask]
 
         nobj = source_type.shape[0]
         d = {
+            "objid": objid.reshape(1, nobj, 1),
             "n_sources": torch.tensor((nobj,)),
             "source_type": source_type.reshape(1, nobj, 1),
-            "plocs": plocs_rv.reshape(1, nobj, 2),
-            "galaxy_fluxes": galaxy_fluxes.reshape(1, nobj, 1),
+            "plocs": plocs.reshape(1, nobj, 2),
+            "galaxy_fluxes": galaxy_fluxes.reshape(1, nobj, 6),
             "galaxy_params": galaxy_params.reshape(1, nobj, 6),
-            "star_fluxes": star_fluxes.reshape(1, nobj, 1),
-            "star_log_fluxes": star_log_fluxes.reshape(1, nobj, 1),
+            "star_fluxes": star_fluxes.reshape(1, nobj, 6),
+            "star_log_fluxes": star_log_fluxes.reshape(1, nobj, 6),
         }
 
         return cls(height, width, d)
 
 
-def get_tile(split_lim, tile_cat):
-    tile_dict = {}
-    param_list = [
-        "locs",
-        "n_sources",
-        "source_type",
-        "galaxy_fluxes",
-        "galaxy_params",
-        "star_fluxes",
-        "star_log_fluxes",
-    ]
-    for i in param_list:
-        tile_dict[i] = tile_cat[i][split_lim[0] : split_lim[1], split_lim[2] : split_lim[3]]
-    return tile_dict
+def read_frame_for_band(image_files, bg_files, n, image_lim):
+    image_list = []
+    bg_list = []
+    wcs = None
+    for b in range(6):
```

Comment:
Better not to hard code 6. Instead, make this function a method and change it to `self.n_bands`, like in the other classes that inherit from `Survey`.